service/src/test/java/bio/terra/profile/service/azure/AzureServiceUnitTest.java

```diff
@@ -0,0 +1,80 @@
+package bio.terra.profile.service.azure;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.profile.app.configuration.AzureConfiguration;
+import bio.terra.profile.common.BaseUnitTest;
+import bio.terra.profile.model.AzureManagedAppModel;
+import com.azure.resourcemanager.managedapplications.models.Application;
+import com.azure.resourcemanager.managedapplications.models.Plan;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+
+public class AzureServiceUnitTest extends BaseUnitTest {
+
+  @Test
+  public void getManagedApps() {
+    var subId = UUID.randomUUID();
+    var authedUserEmail = "profile@example.com";
+    var user =
+        AuthenticatedUserRequest.builder()
+            .setSubjectId("12345")
+            .setEmail(authedUserEmail)
+            .setToken("token")
+            .build();
+    var offerName = "known_terra_offer";
+
+    var authedTerraApp = mock(Application.class);
+    when(authedTerraApp.plan()).thenReturn(new Plan().withProduct(offerName));
```

Comment:
I am not thrilled with the amount of mocking I'm doing here. The alternative is to introduce an adapter class that consumes a com.azure.Application and produces a more testable type. I think this should be ok for now though.